lld/ELF/Arch/AArch64.cpp

```diff
@@ -398,6 +399,10 @@ void AArch64::relocate(uint8_t *loc, const Relocation &rel,
   case R_AARCH64_PREL64:
     write64(loc, val);
     break;
+  case R_AARCH64_AUTH_ABS64:
+    checkIntUInt(loc, val, 32, rel);
```

Comment:
`checkIntUInt` can be an `assert(isInt<32>(val))` thanks to `isInt<32>(sym.getVA(addend))`

Reply:
Sorry, I just realize that this is related to isStaticLinkTimeConstant. See the main comment.

Reply:
Hmm, I just realized `-no-pie` was not handled properly, thanks for bringing attention to this. I've applied the commit 865c9839d0186852f3adaae6ab33551e2ca98818 from your branch, thanks.

P.S. Code here becomes changed after e9b53374522fca221414aac3d5a3b225e7fc73d4